src/test/java/uk/gov/hmcts/reform/userprofileapi/util/JsonFeignResponseHelperTest.java

```diff
@@ -183,4 +175,40 @@ public void privateConstructorTest() throws Exception {
         constructor.setAccessible(true);
         constructor.newInstance((Object[]) null);
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void test_getResponseMapperClass_when_response_success_and_expected_mapper_class_is_passed() {
+        Response response = Response.builder().status(200).reason("OK").body("{\"idamId\": 1}", UTF_8).request(mock(Request.class)).build();
+        Optional<ErrorResponse> optionalObj = getResponseMapperClass(response, ErrorResponse.class);
+        assertTrue(optionalObj.isPresent());
+        assertThat(optionalObj.get()).isEqualTo(ErrorResponse.class);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
```

Comment:
@SupperWarnings also move to the class and remove in the every method

Reply:
done